File: fxgl-base/src/main/java/com/almasb/fxgl/entity/component/ComponentListener.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.entity.component;

/**
 * Notifies when a component is added / removed.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
public interface ComponentListener {

    /**
     * Called after the component was added.
     */
    void onAdded(Component component);

    /**
     * Called before the component is removed.
     */
    void onRemoved(Component component);
}

File: fxgl-base/src/main/kotlin/com/almasb/fxgl/entity/component/ComponentListener.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.entity.component

import com.almasb.fxgl.entity.Component

/**
 * Notifies when a component or a control is added / removed.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
interface ComponentListener {

    fun onAdded(component: Component)

    fun onRemoved(component: Component)
}